sklearn/metrics/_plot/roc_curve.py

```diff
@@ -330,6 +342,12 @@ def from_predictions(
         """
         check_matplotlib_support(f"{cls.__name__}.from_predictions")
 
+        target_type = type_of_target(y_true)
+        if target_type != "binary":
```

Comment:
I do not think we need this check. `roc_curve` calls `_binary_clf_curve` which ends up doing the binary check itself:

https://github.com/scikit-learn/scikit-learn/blob/22ca942536d0e3b2c016174299ff8b23d7335e36/sklearn/metrics/_ranking.py#L736-L738